ors-benchmark/src/main/java/org/heigit/ors/coordinates_generator/generators/CoordinateGeneratorMatrix.java

```diff
@@ -302,82 +343,98 @@ private boolean processSnappedCoordinates(List<double[]> snappedCoordinates, dou
         }
 
         /**
-         * Processes a set of snapped coordinates by verifying distance and routeability constraints,
+         * Processes a set of snapped coordinates by verifying distance and routeability
+         * constraints,
          * and optionally calculating and storing the resulting distance matrix.
          * <p>
          * The method checks for:
          * <ul>
-         *   <li>Maximum allowed distance between row and column coordinates</li>
-         *   <li>Routeability in in forward direction between row elements, and column elements</li>
-         *   <li>Connectors between row and column to create circular routeability</li>
-         *   <li>Computes an asymmetric matrix if all conditions are satisfied</li>
+         * <li>Maximum allowed distance between row and column coordinates</li>
+         * <li>Routeability in forward direction between row elements, and column
+         * elements</li>
+         * <li>Connectors between row and column to create circular routeability</li>
+         * <li>Computes an asymmetric matrix if all conditions are satisfied</li>
          * </ul>
          *
-         * @param snappedCoordinates a list of coordinates, where the first {@code numRows} entries are treated as rows
+         * @param snappedCoordinates a list of coordinates, where the first
+         *                           {@code numRows} entries are treated as rows
          *                           and the next {@code numCols} as columns
-         * @param maxDistance the minimum distance required between each row-column coordinate pair
-         * @return {@code true} if a matrix was successfully computed and processed; {@code false} otherwise
+         * @param maxDistance        the minimum distance required between each
+         *                           row-column coordinate pair
+         * @return {@code true} if a matrix was successfully computed and processed;
+         *         {@code false} otherwise
          */
         private boolean processCoordinatePairs(List<double[]> snappedCoordinates, double maxDistance) {
             List<double[]> row = snappedCoordinates.subList(0, numRows);
             List<double[]> col = snappedCoordinates.subList(numRows, numRows + numCols);
 
-            if (!hasNearbyPoints(row, col, maxDistance)) return false;
-            if (!rowsColsRouteable(row, col)) return false;
-            if (!isRouteable(row.get(0), col.get(0))) return false;
-            if (!isRouteable(row.get(numRows - 1), col.get(numCols - 1))) return false;
+            if (!hasNearbyPoints(row, col, maxDistance))
+                return false;
+            if (!rowsColsRouteable(row, col))
+                return false;
+            if (!isRouteable(row.get(0), col.get(0)))
+                return false;
+            if (!isRouteable(row.get(numRows - 1), col.get(numCols - 1)))
```

Comment:
In order to ensure circular connectivity you should probably check for `isRouteable(col.get(numCols - 1), row.get(numRows - 1)` which corresponds to the connection `D -> 4` in the example from the documentation (the rest of the code already does `(4->3->2->1)->(A->B->C->D)`).